File: pythons/word.py
while True :
    kal = input ("Masukkan kalimat (ketik q untuk keluar): ").lower()
    if kal.lower() in ["q", "quit", "exit", "keluar", "metu"]:
        print("program selesai")
        break
    ter = input ("Masukkan Kata Terlarang (ketik q untuk keluar): ").lower()
    if ter.lower() in ["q", "metu", "keluar", "exit"]:
        print("program selesai")
        break
    char = "-"
    kata = kal.split()
    kata_filtered = [part
                    for katas in kata
                    for part in 
                    (katas.split(char) if char in katas else [word])]
    
    # Menggunakan "in" untuk mengecek apakah suatu string cocok dengan salah satu elemen di string array
    if ter in kata_filtered or kata:
        print ("Kalimat mengandung kata", ter)
    else : 
        print ("Kalimat TIDAK mengandung kata terlarang")
    

File: javas/word.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

public class word {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayList<String> keluarKalimat = new ArrayList<>(Arrays.asList("q", "quit", "exit", "keluar", "metu"));
        ArrayList<String> keluarKata = new ArrayList<>(Arrays.asList("q", "metu", "keluar", "exit"));

        while (true) {
            System.out.print("Masukkan kalimat (ketik q untuk keluar): ");
            String kal = sc.nextLine().toLowerCase();

            if (keluarKalimat.contains(kal)) {
                System.out.println("program selesai");
                break;
            }

            System.out.print("Masukkan Kata Terlarang (ketik q untuk keluar): ");
            String ter = sc.nextLine().toLowerCase();

            if (keluarKata.contains(ter)) {
                System.out.println("program selesai");
                break;
            }

            String[] kata = kal.split("\\s+");
            ArrayList<String> kataFiltered = new ArrayList<>();

            for (String katas : kata) {
                if (katas.contains("-")) {
                    kataFiltered.addAll(Arrays.asList(katas.split("-")));
                } else {
                    kataFiltered.add(katas);
                }
            }

            if (kataFiltered.contains(ter)) {
                System.out.println("Kalimat mengandung kata " + ter);
            } else {
                System.out.println("Kalimat TIDAK mengandung kata terlarang");
            }
        }

        sc.close();
    }
}
